src/mixed-line-bar-chart/bar-series.tsx

```diff
@@ -110,46 +129,86 @@ export default function BarSeries<T extends ChartDataTypes>({
         [styles['series--dimmed']]: dimmed,
       })}
     >
-      {xCoordinates.map(({ x, y, width, height }, i) => {
+      {xCoordinates.map(({ x, y, width, height, isMin, isMax }, i) => {
         if (!isFinite(x) || !isFinite(height)) {
           return;
         }
 
         // Create margins between stacked series but only when series data is not too small.
-        const baseHeightOffset = stackedBarOffsets ? 3 : 0;
+        const baseHeightOffset = isStacked ? 3 : 0;
         const isSmallBar = height < 4;
         const heightOffset = isSmallBar ? 0 : baseHeightOffset;
         const widthOffset = 2;
 
-        const rx = isRefresh ? (isSmallBar ? '2px' : '4px') : '0px';
+        const rx = isRefresh ? (isSmallBar ? 2 : 4) : 0;
+        const placement = isVertical
+          ? {
+              x: x + widthOffset / 2,
+              y: y + heightOffset / 2,
+              width: width - widthOffset,
+              height: height - heightOffset,
+            }
+          : {
+              x: y - height + heightOffset / 2,
+              y: x + widthOffset / 2,
+              width: height - heightOffset,
+              height: width - widthOffset,
+            };
+
         const className = clsx(styles.series__rect, {
           [styles['series--dimmed']]: highlightedXValue !== null && !matchesX(highlightedXValue, series.data[i].x),
         });
+        const styleProps = { fill: color, className };
 
-        return axis === 'x' ? (
-          <rect
-            key={`bar-${i}`}
-            fill={color}
-            x={x + widthOffset / 2}
-            y={y + heightOffset / 2}
-            width={width - widthOffset}
-            height={height - heightOffset}
-            rx={rx}
-            className={className}
-          />
-        ) : (
-          <rect
-            key={`bar-${i}`}
-            fill={color}
-            x={y - height + heightOffset / 2}
-            y={x + widthOffset / 2}
-            width={height - heightOffset}
-            height={width - widthOffset}
-            rx={rx}
-            className={className}
-          />
-        );
+        if (isMin && isMax) {
+          return <rect key={i} {...placement} {...styleProps} rx={rx} />;
+        }
+        if (!isMin && !isMax) {
+          return <rect key={i} {...placement} {...styleProps} rx={0} />;
```

Comment:
Not covered by tests? Why?

Reply:
This part is well covered with screenshot tests. The unit tests do not seem very suitable in this case.

Reply:
😕 this code is a pure math, converting the input data into SVG offsets. See no reason why it is not suitable.

But because this is one line only, I would leave the burden of this tech debt on you to decide